packages/dd-trace/src/llmobs/plugins/openai/utils.js

```diff
@@ -0,0 +1,118 @@
+'use strict'
+
+/**
+ * Extracts chat templates from OpenAI response instructions by replacing variable values with placeholders.
+ *
+ * Performs reverse templating: reconstructs the template by replacing actual values with {{variable_name}}.
+ * For images/files: uses {{variable_name}} when values are available, falls back to [image]/[file] when stripped.
+ *
+ * @param {Array<Object>} instructions - From Response.instructions (array of ResponseInputMessageItem)
+ * @param {Object<string, string>} variables - Normalized variables (output of normalizePromptVariables)
+ * @returns {Array<{role: string, content: string}>} Chat template with placeholders
+ */
+function extractChatTemplateFromInstructions (instructions, variables) {
+  if (!Array.isArray(instructions) || !variables) return []
+
+  const chatTemplate = []
+
+  // Build map of values to placeholders - exclude fallback markers so they remain as-is
+  const valueToPlaceholder = {}
+  for (const [varName, varValue] of Object.entries(variables)) {
+    const valueStr = varValue ? String(varValue) : ''
```

Comment:
maybe we can do this check/normalization in the `normalizePromptVariables` method since it pertains more there (although fine to keep the `{{...}}` part here)